### PR TITLE
[GMMK Pro] Fix unintentional taps to the volume keys when using the encoder

### PR DESCRIPTION
**Pull request by andrebrait**
## Description

In a similar vein to #16830, this fixes the double-tapping or unintentional tapping of the volume keys when using the encoder, introduced in #16721. 

While #16830 took care of fixing the behavior, it still kept the user encoder handling code in the default keymaps. I have removed it, since it's not necessary, given `pro.c` already contains the handling for the media keys (and in an even better way, since it can be more compatible when NKRO is enabled).

I took the time to look for any instances in which user layouts would end up having a different behavior due to #16721. 
Some of them were:

- Introducing a Volume Up/Down key tap where there was none, previously;
  - This happened when the user handled the encoder for other functions in `encoder_update_user` and then returned `true`
- Double-tapping Volume Up/Down due to both the keyboard and user encoder functions both sending taps
  - This happened when the user would send a tap on their `encoder_update_user` implementation and then return `true`

I realize going about changing other people's code is not a _good_ thing, but the code was effectively broken, so here it is.

I also took care of moving an outdated keymap into the `ansi` subdirectory and fixing a small typo in the docs. The whitespace-related changes were done by my text editor and I don't have time to undo them right now. If that's an issue, I'll try to push a new commit later to revert them.

## Sign-off

- [ ] @alexmarmon
- [x] @byungyoonc
- [x] @cedrikl
- [x] @coryginsberg
- [x] @Gigahawk
- [x] @hachetman
- [x] @JackKenney
- [x] @Jonavin (both ansi and iso)
- [x] @mattgauf
- [x] @mike1808
- [x] @stickandgum
- [x] @tguinan
- [x] @ViToni
- [x] @WholesomeDucky
- [x] @willwm
- [x] @zvuc

## Types of Changes

- [ ] Core
- [X] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Issues Fixed or Closed by This PR

* None that have been reported, but it's a better fix for #16832, IMHO.

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
Changing user keymaps requires sign-off from the affected users.

**Comment by andrebrait** [comment]
@zvecr how is this a breaking change?

And ok. Should I add them all on this PR and ask them if they are okay with the change then?

**Comment by zvecr** [comment]
> @zvecr how is this a breaking change?

Its the label we use to manage PRs which touch user code, without prior sign-off. <https://docs.qmk.fm/#/breaking_changes_instructions>.

> Should I add them all on this PR and ask them if they are okay with the change then?

Correct.

**Comment by andrebrait** [comment]
@alexmarmon, @byungyoonc, @cedrikl, @coryginsberg, @gigahawk, @hachetman, @jackkenney, @jonavin (both ansi and iso), @zvuc, @mattgauf, @mike1808, @tguinan, @stickandgum, @wholesomeducky, @willwm, @vitoni

My bad for spamming all of you, but essentially there is a breaking change that affects your keymaps and I'm fixing it. Just need you guys to check if the changes I made reflect the intent here (meaning keeping the behavior your keymaps had _before_ #16721).

**Comment by cedrikl** [comment]
Default mapping for me as well if it breaks I'll fix it at that point. Thanks for letting us know.

File: keyboards/gmmk/pro/ansi/keymaps/mattgauf/keymap.c [hunk 1]
```diff
@@ -173,7 +173,8 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
             }
         }
     }
-    return true;
+    //return true; //set to return false to counteract enabled encoder in pro.c
+    return false;
```
Comment (mattgauf) [inline keyboards/gmmk/pro/ansi/keymaps/mattgauf/keymap.c:177]: Acknowledging with approval, I'll be sure to pull the changes into my fork once the PR is complete.

**Review by mattgauf — APPROVED** [review]

**Comment by WholesomeDucky** [comment]
My keymap uses the encoder for scroll left/right anyway (at least I believe the version on GitHub does) so I'm completely good with this

**Comment by stickandgum** [comment]
I certainly use the encoder but don't know how to verify if this will break any specific functionality. With that said, I trust the community so feel free to modify my code as you see fit.

Thank you.

**Comment by tzarc** [comment]
Added list of required sign-offs to the description.

**Comment by andrebrait** [comment]
> I certainly use the encoder but don't know how to verify if this will break any specific functionality. With that said, I trust the community so feel free to modify my code as you see fit.
> 
> Thank you.

You can pull the code from master and flash to your keyboard. As far as I can tell, before this PR, you'll always get a volume up/down tap in addition to whatever your code is doing. Best case scenario you'd get two taps to the same volume change button. I think you're using your encoder to scroll with mouse buttons. In this case you'd get a volume up/down every time you scrolled up/down.

**Comment by hachetman** [comment]
Hi, also default mapping for me.

**Comment by stickandgum** [comment]
> 

https://github.com/qmk/qmk_firmware/blob/master/keyboards/gmmk/pro/ansi/keymaps/stickandgum/keymap.c#:~:text=%23ifdef%20ENCODER_ENABLE,endif%20//ENCODER_ENABLE
Is the section - I will likely be able to test this over the weekend.

File: keyboards/gmmk/pro/iso/keymaps/vitoni/keymap.c [hunk 2]
```diff
@@ -84,7 +84,8 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
             }
             break;
     }
-    return true;
+    //return true; //set to return false to counteract enabled encoder in pro.c
+    return false;
```
Comment (ViToni) [inline keyboards/gmmk/pro/iso/keymaps/vitoni/keymap.c:88]: LGTM

**Comment by zvuc** [comment]
Just re-compiled firmware with the changed bit and tested encoders. Seems fine!

**Comment by byungyoonc** [comment]
LGTM, thanks for caring!

File: docs/feature_encoders.md [hunk 3]
```diff
@@ -124,7 +124,7 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
 }
 ```
 
-?> Media and mouse countrol keycodes such as `KC_VOLU` and `KC_WH_D` requires `EXTRAKEY_ENABLE = yes` and `MOUSEKEY_ENABLE = yes` respectively in user's `rules.mk` if they are not enabled as default on keyboard level configuration.
+?> Media and mouse control keycodes such as `KC_VOLU` and `KC_WH_D` requires `EXTRAKEY_ENABLE = yes` and `MOUSEKEY_ENABLE = yes` respectively in user's `rules.mk` if they are not enabled as default on keyboard level configuration.
```
Comment (drashna) [inline docs/feature_encoders.md:127]: Please revert. 

This should be opened as a separate PR.

Comment (andrebrait) [reply docs/feature_encoders.md:127]: @drashna done

**Comment by andrebrait** [comment]
@alexmarmon @Gigahawk @JackKenney sorry to bother you, but I still need your blessing here 🙃

**Comment by Gigahawk** [comment]
> @alexmarmon @Gigahawk @JackKenney sorry to bother you, but I still need your blessing here 🙃

LGTM, but I haven't tested it. Haven't had a reason to reflash my board. TY for the fix!

**Comment by andrebrait** [comment]
@drashna @zvecr it seems the only sign-off missing is @alexmarmon, who seems to be active but unresponsive.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/andrebrait/config.h [hunk 4]
```diff
@@ -16,6 +16,10 @@
 
 #pragma once
 
+#ifdef QMK_KEYS_PER_SCAN
+    #undef QMK_KEYS_PER_SCAN
+#endif
+
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/andrebrait/config.h:22]: Deprecated. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
ALso, because this targets user keymaps, this either needs to go to the `develop` branch (you can edit the title to change the target... weird yes, but that's how it handled), or you need each user to sign off on this change.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/andrebrait/config.h (hunk 4 above)
Comment (andrebrait) [reply keyboards/gmmk/pro/rev1/ansi/keymaps/andrebrait/config.h:22]: Was this define deprecated? I'm kinda protecting my own keymap from changes done in the lower precendence `config.h` files.

Comment (tzarc) [reply keyboards/gmmk/pro/rev1/ansi/keymaps/andrebrait/config.h:22]: Yes; all keys are processed in a single iteration of main loop, now.

**Comment by andrebrait** [comment]
> ALso, because this targets user keymaps, this either needs to go to the `develop` branch (you can edit the title to change the target... weird yes, but that's how it handled), or you need each user to sign off on this change.

I switched the base branch to develop and resolved a conflict.

**Comment by andrebrait** [comment]
@drashna let me know if this is good enough now

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]
